pkg/api/types.go

```diff
@@ -1410,6 +1414,54 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// RateLimits is a list of rate limits on disruptions (preemption and other evictions) across the corresponding
+	// collection of pods, e.g. no more than one disruption per hour across the pods of a particular Service
+	RateLimits []RateLimitSpec `json:"rateLimits,omitempty"`
+	// The minium number of pods that must be available simultaneously.
+	// Value can be an absolute number (ex: 5) or a percentage of total pods(ex: 10%).
+	// Absolute number is calculated from percentage by rounding up.
+	// By default, a fixed value of 1 is used.
+	MinAvailable intstr.IntOrString `json:"minAvailable,omitempty"`
```

Comment:
I think using IntOrString is probably not a good idea. Do we have fractions or percentages anywhere else in the API? We have things like millicores but those are integers, not fractions. If we have fractions or percentages somewhere else, use that pattern, otherwise we should use the millicore pattern, like

// exactly one is set
MinAvailableNumberOfInstances *int
MinAvailablePercentageOfInstances *int